pages/companies/[slug]/index.tsx

```diff
@@ -0,0 +1,53 @@
+import axios from 'axios';
+// import { useRouteMatch } from 'react-router-dom';
+import SlugContainer from '../../../components/slug/SlugContainer';
+import CompanyHeader from '../../../components/slug/CompanyHeader';
+import CompanyProfile from '../../../components/slug/CompanyProfile';
+import CompanyContacts from '../../../components/slug/CompanyContacts';
+import QuantityAdminsPartners from '../../../components/slug/QuantityAdminsPartners';
+import TableRender from '../../../components/slug/TableRender';
+import PreviousInfo from '../../../components/slug/CompanyPreviousInfo';
+import CompanyInvestCapital from '../../../components/slug/CompanyInvestCapital';
+import SimilarCompanies from '../../../components/slug/SimilarCompanies';
+import CompanyTurnover from '../../../components/slug/CompanyTurnover';
+import { ISlug } from '../../../types/slug';
+import styles from '../../../styles/Home.module.css';
+import Loading from '../../../components/Loading';
+
+axios.defaults.baseURL = 'https://app.informer.md/api/public';
+
+function CompanyId({ slug }: ISlug) {
+  // const {url, path} = useRouteMatch();
+  // console.log(url);
+  // console.log(path);
+  if (!slug) {
+    return <Loading />;
+  }
+  return (
+    <div className={styles.wrapper}>
+      <CompanyHeader slug={slug} />
```

Comment:
It is not best practice to pass data through props and we will implement all other situations:
- Create 2 similar pages like: 
```javascript
    pages/companies/deprecated/[slug]/index.tsx  // Migrate your current variant here
    pages/companies/context/[slug]/index.tsx  // Save data to `AppContext`
    pages/companies/react-query/[slug]/index.tsx // Prefetch query at initial props

// Install `react-query` and setup `queryClient` in your `_app.tsx`
function MyApp({ Component, pageProps }) {
    const [queryClient] = useState(() => new QueryClient({
      defaultOptions: {
        queries: {
          refetchOnWindowFocus: false,
          refetchOnMount: true,
        },
      },
    }));

    return (
        <QueryClientProvider client={queryClient}>
            <Hydrate state={pageProps.dehydratedState}>
                ...

// Add SSR util to get dehydrate state of queries: utils/ssr.ts
import { dehydrate, QueryClient } from 'react-query';

export const getQueries = (queryClient: QueryClient): Record<string, any> =>
  dehydrate(queryClient).queries.reduce(
    (prevState, { state: { data }, queryHash }) => ({
      ...prevState,
      [JSON.parse(queryHash)[0]]: data, // Return data by query key
    }),
    {},
  );

// After this steps we can prefetch query in getInitialProps
Page.getInitialProps = async ({ query }): Promise<Record<string, any>> => {
  const queryClient = new QueryClient(defaultOptions);

  await queryClient.prefetchQuery(
    ['data', { id: query?.slug }],
    ({ queryKey: [, query] }: Record<string, any>) => axios.get(
      `/company?slug=${query.slug}`
    ).then(({ data }) => data), // It would be better to move requests to a separate api folder 
    { cacheTime },
  );

  return getQueries(queryClient); // Returns: { data }
};
```